tracee-ebpf/tracee/events_pipeline.go

```diff
@@ -165,24 +165,34 @@ func (t *Tracee) processEvents(done <-chan struct{}, in <-chan *external.Event)
 				continue
 			}
 
+			// Derive event before parsing its arguments
+			derivatives := t.deriveEvent(*event)
```

Comment:
just a question - why passing as value and not by reference?

Reply:
deriveEvent shuold alter the original event.
In addition, there is no performance gain in golang to pass by pointer (even the opposite on small variables)